lib/cocoapods/external_sources/abstract_external_source.rb

```diff
@@ -116,10 +116,12 @@ def pre_download(sandbox)
           rescue Pod::DSLError => e
             raise Informative, "Failed to load '#{name}' podspec: #{e.message}"
           rescue => _
-            raise Informative, "Unable to find a specification for '#{name}'."
+            raise Informative, "Failed to download '#{name}'."
           end
           spec = download_result.spec
 
+          raise Informative, "Unable to find a specification for '#{name}'." unless spec
```

Comment:
There is a possibility that the pre-download "works" but the podspec is not found in the source. This should not have been removed!